### PR TITLE
assertions for Student, University and Allocator.

### PR DESCRIPTION
**Pull request by paulktham**

### CODE REVIEW

**Comment by isaacsaw25** [comment]
It seems your assertions have caused a failure in the CI test. Perhaps re-look at the assertion logic to ensure correctness? 😅

File: src/main/java/university/University.java [hunk 1]
```diff
@@ -35,6 +39,7 @@ public String getAcronym() {
     }
 
     public int getSpotsLeft() {
+        assert spotsLeft > 0 : "Cannot remove a spot when none are left";
```
Comment (isaacsaw25) [inline src/main/java/university/University.java:42]: The logic here may be wrong, perhaps try `>=0`? 😅 🥶

**Review by isaacsaw25 — CHANGES_REQUESTED** [review]
Some minor adjustments needed to get the checks working again! 화이팅!

File: src/main/java/university/University.java (hunk 1 above)
Comment (paulktham) [reply src/main/java/university/University.java:42]: oh goodness me! I placed that at the wrong function!

**Review by isaacsaw25 — APPROVED** [review]
Great work on the fix! LGTM 👍 😄

**Pull request merged**